### PR TITLE
Release 4.0.1

### PR DESCRIPTION
**Pull request by repeatedly**
Update fluentd and plugin gems.

### CODE REVIEW

**Comment by repeatedly** [comment]
For msi build on github actions, do we need to update Windows Gemfile.lock?

File: gemfiles/linux/Gemfile [hunk 1]
```diff
@@ -4,59 +4,62 @@ source "https://rubygems.org"
...
   gem "ffi-win32-extensions", "1.0.3"
-  gem "fluent-plugin-parser-winevt_xml", "0.2.2"
-  gem "fluent-plugin-windows-eventlog", "0.7.0"
   gem "win32-api", "1.9.1"
   gem "win32-event", "0.6.3"
-  gem "win32-eventlog", "0.6.7"
-  gem "win32-ipc", "0.6.1"
-  gem "win32-service", "0.8.3"
+  gem "win32-ipc", "0.7.0"
+  gem "win32-service", "2.1.5"
```
Comment (cosmo0920) [inline gemfiles/linux/Gemfile:55]: Fluentd gempsec cannot handle these newer gems by bundler:
https://github.com/fluent/fluentd/blob/master/fluentd.gemspec#L35-L36
If we should bump up version for these gems, we should upgrade dependent gem versions on Fluentd's gemspec. 
This causes package building error on Windows.

**Comment by cosmo0920** [comment]
> For msi build on github actions, do we need to update Windows Gemfile.lock?

Yes. For rpm/apt building on GitHub Actions, we also need to update Gemfile.lock for Linux with:
```
$ cd td-agent && rake lockfile:update
```

**Comment by cosmo0920** [comment]
I'd registered to fix win32 related version conflicts issue on Windows:
https://github.com/fluent/fluentd/pull/3100
This issue is caused by bundler based migration....

**Comment by repeatedly** [comment]
> Should we bump up version to v4.0.1?

I updated version locally for testing.
We want to fix https://github.com/fluent-plugins-nursery/td-agent-builder/issues/176 for this release, so I didn't push the changes yet.

**Comment by cosmo0920** [comment]
Oh. I got it.

**Comment by cosmo0920** [comment]
#177 is merged. Could you rebase off master?

**Pull request merged**